src/esm_parser/esm_parser.py

```diff
@@ -2413,14 +2410,18 @@ def do_math_in_entry(tree, rhs, config):
                     math = math + "all_dates[" + str(index) + "]"
                     index += 1
         result = eval(math)
-        if isinstance(result, list):
-            if keep_list:
-                return ListWithProvenance(result, [])
+        if isinstance(result, list) and date_operation:
             result = result[
                 -1
             ]  # should be extended in the future - here: if list (= if diff between dates) than result in seconds
+        elif isinstance(result, list):
+            entry = ListWithProvenance(result, None)
+            entry.set_provenance(rhs.provenance)
```

Comment:
I might not understand something about provenance 100% yet, but why not this version:
```suggestion
            entry = ListWithProvenance(result, rhs.provenance)
```

Reply:
For giving a provenance during instanciation of `ListWithProvenance` (or `DictWithProvenance`) the provenance needs to have the same structure as the original list or dict. That is to say, if you have a `dict` with some key-values, what you need to give in the second argument of the instancing is also a `dict` with the same keys, but the values are provenances. `set_provenance` is used to assign the same unique provenance to a whole list/or dict with provenance, which is what we want in this case (we are expanding a string into a list so we want to assign a single provenance information to the new elements of the list.